src/components/plot/bookmarkbutton.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {FacetedCompositeUnitSpec} from 'vega-lite/build/src/spec';
+import {BOOKMARK_ADD_PLOT, BOOKMARK_REMOVE_PLOT, BookmarkAction} from '../../actions/bookmark';
+import {ActionHandler} from '../../actions/redux-action';
+import {Bookmark} from '../../models/bookmark';
+import {PlotFieldInfo} from '../../models/plot';
+import * as styles from './bookmarkbutton.scss';
+
+
+
+export interface BookmarkProps extends ActionHandler<BookmarkAction> {
+  bookmark: Bookmark;
+  plotObjectFieldInfos: PlotFieldInfo[];
+  plotObjectSpec: FacetedCompositeUnitSpec;
+}
+
+export class BookmarkButtonBase extends React.PureComponent<BookmarkProps, any> {
+  constructor(props: BookmarkProps) {
+    super(props);
+
+    this.state = {
```

Comment:
Don't need it for `isBookmarked` -- moved to inside render. 

openDialog will be removed one I rewrite the bookmarkAlert as a tether component.